sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -64,21 +64,18 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                 throw new InvalidOperationException("Bearer token authentication is not permitted for non TLS protected (https) endpoints.");
             }
 
-            if (DateTimeOffset.UtcNow >= _refreshOn)
+            (string? headerValue, bool scheduleRefreshToken) = await _accessTokenCache.GetHeaderValueAsync(async, message.CancellationToken);
+            if (headerValue == null)
             {
-                AccessToken token = async ?
-                        await _credential.GetTokenAsync(new TokenRequestContext(_scopes, message.Request.ClientRequestId), message.CancellationToken).ConfigureAwait(false) :
-                        _credential.GetToken(new TokenRequestContext(_scopes, message.Request.ClientRequestId), message.CancellationToken);
-
-                _headerValue = "Bearer " + token.Token;
-                _refreshOn = token.ExpiresOn - TimeSpan.FromMinutes(2);
+                headerValue = await GetHeaderValueFromCredentialAsync(message, async);
             }
-
-            if (_headerValue != null)
+            else if (scheduleRefreshToken)
             {
-                message.Request.SetHeader(HttpHeader.Names.Authorization, _headerValue);
+                _ = Task.Run(() => GetHeaderValueFromCredentialAsync(message, true));
```

Comment:
@pakrym Do we have any better way to handle this task than make it just hanging in the void?

Reply:
What happens if this task fails? Would we ever try again before the token fully expires or when it does expire?

Reply:
If it fails, `_accessTokenCache.ResetTokenIfExpired` will be called. If token has expired while `TokenCredential.GetTokenAsync` was executed, `AccessTokenCache` will be set into `Invalid` state. If it is still not expired, it will be put back into `Valid` state so that next request can trigger another update attempt.